### PR TITLE
Capybara & N+1 problem

### PR DESCRIPTION
**Pull request by gjuliao**
## Requirement
- Paste before and after

Before:
Pasted Graphic 3

After:
Pasted Graphic 4

### CODE REVIEW

File: spec/views/users/show.html.erb_spec.rb [hunk 1]
```diff
@@ -1,5 +1,68 @@
...
+    expect(page).to have_content(/posts/i)
+  end
+
+  it 'Can see user bio' do
+    expect(page).to have_text(user.bio)
+  end
+
+  it 'Can see button with text See All Posts' do
+    expect(page).to have_text(/See All Posts/i)
+  end
```
Comment (franclobo) [inline spec/views/users/show.html.erb_spec.rb:62]: - [ ] Well done on the tests for user show page, please add all tests required to approve the milestone 👇🏽 

> Tests that assert you can see the user's first 3 posts.
> Tests that redirect to the post's show page when you click on a user's post.

File: spec/views/posts/index.html.erb_spec.rb [hunk 2]
```diff
@@ -1,5 +1,107 @@
...
+    expect(page).to have_content(post3.title)
+  end
+
+  it 'Checking for correct post text' do
+    expect(page).to have_content(post3.text)
+  end
+
+  it 'Checking for comment in post' do
+    expect(page).to have_content('Comments: 1')
+  end
```
Comment (franclobo) [inline spec/views/posts/index.html.erb_spec.rb:93]: - [ ] Well done implementing the tests, please 🙏🏽 add all required tests to approve the milestone 👇🏽 

> Tests that assert you can see the first comments on a post.

**Review by franclobo — CHANGES_REQUESTED** [review]
Hi @gjuliao and @PrangonGhose, 

# ♻️ Required Changes ♻️

so-close-jim-halpert

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

- No linter errors ✔️ 
- Correct Git flow ✔️ 
- Descriptive PR ✔️ 
- All tests passed ✔️
- N + 1 problems fixed ✔️ 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- [ ] If you are struggling with tests, here's a [capybara cheat sheet](https://gist.github.com/zhengjia/428105) that can help unblock you.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by jerryowusu — CHANGES_REQUESTED** [review]
Hi Team, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


so-close


## Highlights
Some of the tests are passing :heavy_check_mark: 
Used capybara gem :heavy_check_mark: 

## Required Changes ♻️

- [ ] The posts_spec.rb tests are not working (failure/error) because there is a syntax error in  line 62. You failed to provide value for the `author` as shown below. Kindly check and fix it so that all the tests will run and pass since it is a project requirement. 

Screenshot from 2023-04-11 15-54-29

Screenshot from 2023-04-11 16-02-23

## Optional suggestions

- [ ] Kindly consider adding your partner's credentials to the readme section as an author since it is a pair programming exercise

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by julie-ify — COMMENTED** [review]
Hi @gjuliao and @PrangonGhose,

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

### Kindly fix your linter errors

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Review by pinkmoon25 — APPROVED** [review]
Hi @gjuliao @PrangonGhose , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
Rubocop checks :heavy_check_mark: 
gitflow :heavy_check_mark: 
Tests fixed :heavy_check_mark: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**